Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,438 @@
+		/// </summary>
+		/// <param name="externalId">Platform-specific calendar identifier</param>
+		/// <returns>The corresponding calendar, or null if not found</returns>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<Calendar> GetCalendarByIdAsync(string externalId)
+		{
+			Calendar calendar = null;
+			CalendarRecord calRecord = null;
+
+			try
+			{
+				calendar = new Calendar();
+				calRecord = new CalendarRecord(Book.Uri);
+				calRecord.Set<string>(Book.Name, "org.tizen.calendar");
+				calendar.Name = calRecord.Get<string>(Book.Name);
+				calendar.ExternalID = calRecord.Get<int>(Book.Id).ToString();
+				calendar.CanEditEvents = true;
+				calendar.CanEditCalendar = false;
+				calendar.AccountName = calRecord.Get<int>(Book.AccountId).ToString();
+				calendar.Color = calRecord.Get<string>(Book.Color);
+			}
+			finally
+			{
+				calRecord?.Dispose();
+				calRecord = null;
+			}
+			return Task.FromResult(calendar);
+		}
+
+		/// <summary>
+		/// Gets all events for a calendar within the specified time range.
+		/// </summary>
+		/// <param name="calendar">Calendar containing events</param>
+		/// <param name="start">Start of event range</param>
+		/// <param name="end">End of event range</param>
+		/// <returns>Calendar events</returns>
+		/// <exception cref="System.ArgumentException">Calendar does not exist on device</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTime start, DateTime end)
+		{
+			var calEvent = new List<CalendarEvent>();
+			CalendarManager calManager = null;
+			CalendarQuery calQuery = null;
+			CalendarList calList = null;
+			CalendarRecord calRecord = null;
+			CalendarFilter calFilter = null;
+
+			try
+			{
+				calManager = new CalendarManager();
+				calFilter.AddCondition(CalendarFilter.LogicalOperator.And, Event.Start, CalendarFilter.IntegerMatchType.GreaterThanOrEqual, ConvertIntPtrToCalendarTime(start));
+				calFilter.AddCondition(CalendarFilter.LogicalOperator.And, Event.End, CalendarFilter.IntegerMatchType.LessThanOrEqual, ConvertIntPtrToCalendarTime(end));
+				calQuery = new CalendarQuery(Event.Uri);
+				calQuery.SetFilter(calFilter);
+				calList = calManager.Database.GetRecordsWithQuery(calQuery, 0, 0);
+
+				if (calList.Count == 0)
+					return Task.FromResult<IList<CalendarEvent>>(calEvent);
+
+				calList.MoveFirst();
+
+				do
+				{
+					calRecord = calList.GetCurrentRecord();
+					string summary = calRecord.Get<string>(Event.Summary);
+					CalendarTime startTime = calRecord.Get<CalendarTime>(Event.Start);
```

Comment:
This throws an exception for all-day events:

```
Got exception: System.ArgumentOutOfRangeException: Year, Month, and Day parameters describe an un-representable DateTime.
   at System.DateTime.DateToTicks(Int32 year, Int32 month, Int32 day)
   at Tizen.Pims.Calendar.CalendarRecord.ConvertIntPtrToCalendarTime(DateTime time)
   at Tizen.Pims.Calendar.CalendarRecord.Get[T](UInt32 propertyId)
```
No idea what the problem is or how to fix it... considering that it's in the Tizen code and an all-day event still has to know _what day it is_...
:(